src/styles/components/_details.scss

```diff
@@ -0,0 +1,140 @@
+@import '../settings/colors';
+@import '../settings/global';
+@import '../settings/type';
+@import '../tools/mixins';
+
+$CHEVRON_SIZE: 24px;
+$BORDER_COLOR: $GREY_300;
+
+.w-details {
+  border: 1px solid $BORDER_COLOR;
+  border-width: 1px 0;
+  display: flex;
+  flex-wrap: wrap;
+  margin: 36px 0;
+  padding: 18px 0;
+  position: relative;
+
+  & + & {
+    border-top: 0;
+    margin-top: -36px;
+  }
+}
+
+.w-details__summary {
+  cursor: pointer;
+  flex-grow: 1;
+  font: inherit;
+  list-style: none;
+
+  // scss-lint thinks this is a pseudo class instead of a psuedo element
+  // and complains about the two colons.
+  // scss-lint:disable all
+  &::-webkit-details-marker {
+    display: none;
+  }
+  // scss-lint:enable all
+
+  &:focus {
+    outline: none;
+  }
+
+  &:hover .w-details__header::after,
+  &:focus .w-details__header::after {
+    background: $LIGHT_BLUE_50;
+  }
+
+  &:focus .w-details__header::after {
+    box-shadow: 0 0 0 1px $FOCUS_COLOR;
+  }
+
+  &:active .w-details__header::after {
+    background: $LIGHT_BLUE_100;
+  }
+}
+
+// The details header class overrides the margins of the base element and
+// adds padding so that the tap target is sufficiently tall.
+.w-details__header {
+  color: $LIGHT_BLUE_600;
+  margin: 0;
+  padding: .75rem #{$CHEVRON_SIZE + 8px} .75rem 0;
+  position: relative;
+
+  &::after {
+    @include font-material-icon();
+    border-radius: 50%;
+    content: 'expand_more';
+    cursor: $CURSOR_POINTER_VALUE;
+    font-size: $CHEVRON_SIZE;
+    padding: 2px;
+    position: absolute;
+    right: 0;
+    speak: none;
+    top: 50%;
+    transform: translateY(-50%);
+    transition: background .2s, transform .3s;
+    visibility: visible;
+  }
+
+  /// adds the breakpoint styles
+  @include bp(xsm) {
+    &::after {
+      padding: .25em;
+    }
+  }
+}
+
+.w-details[open] .w-details__header::after {
+  transform: rotate(180deg) translateY(50%);
+}
+
+// The following selectors modulate the vertical whitespace of (a) headers (when
+// there's no accompanying preview) and (b) previews in both open and closed
+// states. They do violate BEM, but using helper classes would add a fair
+// amount of complexity to the component JS.
+
+.w-details__header:only-child {
+  line-height: 1.5rem;
+  padding: .375rem #{$CHEVRON_SIZE + 8px} .375rem 0;
+}
+
+h2.w-details__header:only-child {
```

Comment:
Just for h2. What's happening here:
- I'm adjusting vertical whitespace so it looks OK with standalone summaries (i.e., summaries without a preview). That's only needed for h3-h6, which is what the preceding override does.
- Since h2's spacing is fine as-is, I have to override the override.